### ISSUE TITLE
Switch Idea Hub module REST routes to actually use Idea Hub API

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#3518]
Now that the Idea Hub API client library code is available, Site Kit should actually start using the API.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `Idea_Hub` class should have its REST data points that are intended to proxy through requests to the Idea Hub API updated to actually use the API. **The necessary PHP client service classes are included in Site Kit via #3517.**
    * The class needs to implement the `setup_services` method, to bootstrap a single `ideahub` service, which should be [this class](https://github.com/googleapis/google-api-php-client-services/blob/master/src/Google/Service/Ideahub.php).
    * The `get_datapoint_definitions` method needs to be modified to reference the new `ideahub` service for all data points that use the API (`GET:new-ideas`, `GET:saved-ideas`, `POST:update-idea-state`).
    * The three above endpoints need to be connected to the respective Idea Hub API endpoint:
        * For `GET:new-ideas` and `GET:saved-ideas`, remove the hard-coded return values and instead use https://github.com/googleapis/google-api-php-client-services/blob/master/src/Google/Service/Ideahub/Resource/PlatformsPropertiesIdeas.php, passing `sitekit` for platform and the _URL-encoded_ current site's URL ("reference site URL") for property. In order to only return new ideas, the `filter` option needs to be used, passing a string "saved(false)". In order to only return saved ideas, the `filter` option needs to be used, passing a string "saved(true)".
        * ~~For `POST:update-idea-state`, use https://github.com/googleapis/google-api-php-client-services/blob/master/src/Google/Service/Ideahub/Resource/PlatformsPropertiesIdeaStates.php, passing platform and property as mentioned above and the received idea name as the "ideaState" placeholder in the path. The data point should expect a single variable `ideaState` which needs to be an object / associative array with `name` (idea name string, also used in the path as mentioned right above), `saved` (boolean) and `dismissed` (boolean) fields.~~

## Implementation Brief

* Add a `protected function setup_services` to the `Idea_Hub` class that creates a new instance of Google_Service_Ideahub` to be used in the instance of `Idea_Hub`.
* Modidy the existing `get_datapoint_definitions` in the `Idea_Hub` class so that the following point to `array( 'service' => 'ideahub' )`:
    * `GET:new-ideas`
    * `GET:saved-ideas`
    * ~~`POST:update-idea-state`~~
* See ACs for the implementation of each endpoint, as the ACs outline the implementation quite well.

### Test Coverage

* Existing tests should pass, but no new tests needed.

### Visual Regression Changes

* N/A

## QA Brief

* Activate Idea Hub module
* Set the custom site URL setting in the Site Kit developer settings to `https://www.elasticpress.io/`
* Go to the dashboard and verify that you see new ideas
* Also check that you can create a draft post using a new idea and that idea won't re-appear in the new ideas list once you refresh the dashboard page.

## Changelog entry

* Update Idea Hub to use the production API.

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by felixarntz** [comment]
@eugene-manuilov Just a tiny update here: Based on a few updates from the API team, it looks like the value we need to pass for the `platform` is "sitekit" instead of "SITEKIT" what I had in the ACs earlier. If one doesn't work, please try out the other, just to make sure :)

**Comment by felixarntz** [comment]
@eugene-manuilov One more thing affecting this has been clarified by the API team just now, which I've updated in the ACs:
> In order to only return new ideas, the filter option needs to be used, passing a string "saved(false)". In order to only return saved ideas, the filter option needs to be used, passing a string "saved(true)".

So other than I assumed before, we will need to pass `saved(false)` to only get new ideas back.

**Comment by eugene-manuilov** [comment]
@felixarntz, unfortunately, it doesn't seem to work... I get 404 errors for all combinations. Here is what I have:

```php
$reference_url = $this->context->get_reference_site_url();
$reference_url = rawurlencode( $reference_url );

$parent = "publishers/sitekit/properties/{$reference_url}";

return $this->get_service( 'ideahub' )->platforms_properties_ideas->listPlatformsPropertiesIdeas( $parent, array(
    'filter' => 'saved(false)',
) );
```

It throws an exception with 404 status code and HTML code in the message that renders into this:

![sitekit 10uplabs com_wp-json_google-site-kit_v1_modules_idea-hub_data_new-ideas__locale=user](https://user-images.githubusercontent.com/873049/123279179-4a705180-d510-11eb-94d1-f16170e491be.png)

Am I missing something? :thinking:

**Comment by felixarntz** [comment]
@eugene-manuilov Ohhh wow - I think the API documentation is incorrect there. Now I realize why I was confused before between the term "platforms" and "publishers". So, despite of what the docs are saying, you need to use `platforms/sitekit/...` instead of `publishers/sitekit/...`. When I open that URL, I don't get a 404. Sorry about the confusion here.

**Comment by eugene-manuilov** [comment]
Thanks, @felixarntz. Seems to work now, but I get a new error: 

```json
{
    "code": 403,
    "message": "Idea Hub API has not been used in project 609871864289 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/ideahub.googleapis.com/overview?project=609871864289 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.",
    "data": {
        "status": 403,
       "reason": "accessNotConfigured"
     }
}
```

Did we forget to activate it in the developers console?

**Comment by felixarntz** [comment]
@eugene-manuilov 🤦‍♂️ Apologies, we had only activated it in the staging version so far. Can you try again now?

**Comment by eugene-manuilov** [comment]
> @eugene-manuilov  Apologies, we had only activated it in the staging version so far. Can you try again now?

Seems to work now, but the only answer I get is: 
```json
{
    "nextPageToken": null
}
```

If I use a different domain, it always returns the error response:

```json
{
    "code": 500,
    "message": "Internal error encountered.",
    "data": {
        "status": 500,
        "reason": "backendError"
    }
}
```

**Comment by felixarntz** [comment]
Thanks @eugene-manuilov - which domain did you use when you got the `nextPageToken` response? And which one for the 500 error? In case you're okay to share this.

I'll bring this up with the API team.

**Comment by eugene-manuilov** [comment]
When I use `http://sitekit.10uplabs.com`, it returns the response with the `nextPageToken` property. And when I used a random `http://mysite.com`, it returns the internal server error.

**Comment by felixarntz** [comment]
@eugene-manuilov Per our Slack conversation, assigning this back to you for further investigation/testing.

**Comment by eugene-manuilov** [comment]
@felixarntz can you help me to test my PR #3681? Unfortunately, when I use it on my local machine, I receive no ideas... If I use the site that you gave me, I get 403 responses...

**Comment by felixarntz** [comment]
@eclarke1 @fhollis @eugene-manuilov Quick update here: Unfortunately this is still blocked by the API not returning ideas. So we may need to postpone the bug bash further.

**Comment by felixarntz** [comment]
@eugene-manuilov I now get ideas back for some of the sites we requested addition to the pipeline for. Can you please try again with one of them?

I now see the ideas in the API response, your PR doesn't work just yet (it causes an error due to https://github.com/google/site-kit-wp/pull/3681#discussion_r669105621) but you can iterate on that once you get it to work.

**Comment by felixarntz** [comment]
@eugene-manuilov Per our Slack conversation, I've reduced the scope of this issue to no longer include any changes to `POST:update-idea-state`. This is now broken out into #3747.

**Comment by eugene-manuilov** [comment]
@felixarntz could you please check my PR? It contains implementation for new-ideas and saved-ideas endpoints. Plus it includes modifications in the idea hub datastores that are required to make sure selectors work correctly for the Idea Hub widget.

**Comment by aaemnnosttv** [comment]
@eugene-manuilov would you please add a QAB?

**Comment by eugene-manuilov** [comment]
@wpdarren this is ready for QA.

**Comment by wpdarren** [comment]
### QA Update:  Pass ✅ 

**Verified:**

- With the custom site URL as `elasticpress.io` the dashboard shows new ideas  - [Screenshot](https://user-images.githubusercontent.com/73545194/128760561-8a6a09d2-167f-4186-864e-d1a8edd8a432.png)
- Confirmed that you can create a draft post using a new idea - [Screenshot](https://user-images.githubusercontent.com/73545194/128760694-31a0f34d-a93d-41d9-9812-3711337bdcf6.png)
- Confirmed that on refresh the idea does not re-appear in the new ideas list.
